files/zh-cn/web/api/web_speech_api/using_the_web_speech_api/index.md

```diff
@@ -267,30 +263,32 @@ var voices = [];
 
 #### 填充 select 元素
 
-为使用设备可使用的不同的语音选项填充`<select>`元素，我们写了`populateVoiceList()` 方法。首先调用[`SpeechSynthesis.getVoices()`](/zh-CN/docs/Web/API/SpeechSynthesis/getVoices) ，这个函数返回包含所有可用语音 ([`SpeechSynthesisVoice`](/zh-CN/docs/Web/API/SpeechSynthesisVoice)对象) 的列表。接下来循环这个列表，每次创建一个`<option>` 元素，设置它的文本内容以显示声音的名称（从`SpeechSynthesisVoice.name`获取），语音的语言（从`SpeechSynthesisVoice.lang`获取），如果某个语音是合成引擎默认的 (检查[`SpeechSynthesisVoice.default`](/zh-CN/docs/Web/API/SpeechSynthesisVoice/default)属性返回`true`) 在文本内容后面添加`-- DEFAULT`。
+为使用设备上可用的不同的语音选项填充 {{htmlelement("select")}} 元素，我们写了一个 `populateVoiceList()` 方法。首先调用 [`SpeechSynthesis.getVoices()`](/zh-CN/docs/Web/API/SpeechSynthesis/getVoices) ，这个函数返回包含所有可用语音 ([`SpeechSynthesisVoice`](/zh-CN/docs/Web/API/SpeechSynthesisVoice)对象) 的列表。接下来循环这个列表，每次创建一个 {{htmlelement("option")}} 元素，设置它的文本内容以显示声音的名称（从 {{domxref("SpeechSynthesisVoice.name")}} 获取），语音的语言（从 {{domxref("SpeechSynthesisVoice.lang")}} 获取），如果某个语音是合成引擎默认的 (检查 {{domxref("SpeechSynthesisVoice.default")}} 为 `true` 的属性) 在文本内容后面添加 `-- DEFAULT`。
 
-对于每个`option`元素，我们也创建了`data-` 属性，属性值是语音的名字和语言，这样在之后我们可以轻松获取这个信息。之后把所有的`option`元素作为孩子添加到`select` 元素。
+对于每个 `option` 元素，我们也创建了 `data-` 属性，属性值是语音的名字和语言，这样在之后我们可以轻松获取这个信息。之后把所有的 `option` 元素作为孩子添加到 `select` 元素内。
 
 ```js
 function populateVoiceList() {
   voices = synth.getVoices();
 
-  for (i = 0; i < voices.length; i++) {
-    var option = document.createElement("option");
-    option.textContent = voices[i].name + " (" + voices[i].lang + ")";
+  for (const voice of voices) {
+    const option = document.createElement("option");
+    option.textContent = `${voice.name} (${voice.lang})`;
 
-    if (voices[i].default) {
-      option.textContent += " -- DEFAULT";
+    if (voice.default) {
+      option.textContent += " — DEFAULT";
     }
 
-    option.setAttribute("data-lang", voices[i].lang);
-    option.setAttribute("data-name", voices[i].name);
+    option.setAttribute("data-lang", voice.lang);
+    option.setAttribute("data-name", voice.name);
     voiceSelect.appendChild(option);
   }
 }
 ```
 
-接下来是运行这个函数，我们做如下代码工作。因为 Firefox 不支持[`SpeechSynthesis.onvoiceschanged`](/zh-CN/docs/Web/API/SpeechSynthesis/onvoiceschanged) ，所以很常规地只是返回语音对象列表当[`SpeechSynthesis.getVoices()`](/zh-CN/docs/Web/API/SpeechSynthesis/getVoices) 被触发。但是 Chrome 就有点不同了，在`SpeechSynthesis.getVoices()` 被触发时，先要等待事件触发 (有点绕\~按照下面代码，`populateVoiceList` 函数在 Firefox 运行一次，在 Chrome 中运行两次)：
+早期版本的浏览器不支持 {{domxref("SpeechSynthesis.voiceschanged_event", "voiceschanged")}} 事件，只有当 {{domxref("SpeechSynthesis.getVoices()")}} 被触发时才返回语音列表。
+而其他浏览器，比如 Chrome 中，你必须等待 `voiceschanged` 事件触发后才能获得可用语音列表。
+为了兼容这两种情况，我们运行如下代码：
```

Comment:
Do not break lines, which will introduce spaces.

```suggestion
早期版本的浏览器不支持 {{domxref("SpeechSynthesis.voiceschanged_event", "voiceschanged")}} 事件，只有当 {{domxref("SpeechSynthesis.getVoices()")}} 被触发时才返回语音列表。而其他浏览器，比如 Chrome 中，你必须等待 `voiceschanged` 事件触发后才能获得可用语音列表。为了兼容这两种情况，我们运行如下代码：
```

Reply:
The [English version](https://github.com/mdn/content/blob/main/files/en-us/web/api/web_speech_api/using_the_web_speech_api/index.md?plain=1#L271-L273) looks good with multiple lines. Is there any possibility to remove the spaces from the rendering engine? Having all contents in one line makes the markdown file difficult to maintain.

Reply:
No, it is markdown spec for line breaks to introduce whitespace. Doing this is not correct.